library/accessory_workflows/build_cellranger_reference/BuildCellRangerReferenceBundle.wdl

```diff
@@ -0,0 +1,80 @@
+task BuildCellRangerReference {
+  String ref_fasta
+  String gtf_file
+  String ref_name
+
+  String docker = "quay.io/humancellatlas/secondary-analysis-cellranger"
+  String memory = "416 GB"
+  Int boot_disk_size_gb = 12
+  String disk_space = "250"
+  Int cpu = 64
+
+  command {
+    set -e
+
+    wget ${ref_fasta}
+    ref_fasta_tar_filename=$(echo ${ref_fasta} | sed 's:.*/::')
+    gunzip $ref_fasta_tar_filename
+
+    ref_fasta_filename=$(echo "$ref_fasta_tar_filename" | rev | cut -c4- | rev)
+
+    wget ${gtf_file}
+    gtf_tar_filename=$(echo ${gtf_file} | sed 's:.*/::')
+    gunzip $gtf_tar_filename
+
+    gtf_filename=$(echo "$gtf_tar_filename" | rev | cut -c4- | rev)
+
+    cellranger mkgtf $gtf_filename "filtered.$gtf_filename" \
+                     --attribute=gene_biotype:protein_coding \
+                     --attribute=gene_biotype:lincRNA \
+                     --attribute=gene_biotype:antisense \
+                     --attribute=gene_biotype:IG_LV_gene \
+                     --attribute=gene_biotype:IG_V_gene \
+                     --attribute=gene_biotype:IG_V_pseudogene \
+                     --attribute=gene_biotype:IG_D_gene \
+                     --attribute=gene_biotype:IG_J_gene \
+                     --attribute=gene_biotype:IG_J_pseudogene \
+                     --attribute=gene_biotype:IG_C_gene \
+                     --attribute=gene_biotype:IG_C_pseudogene \
+                     --attribute=gene_biotype:TR_V_gene \
+                     --attribute=gene_biotype:TR_V_pseudogene \
+                     --attribute=gene_biotype:TR_D_gene \
+                     --attribute=gene_biotype:TR_J_gene \
+                     --attribute=gene_biotype:TR_J_pseudogene \
+                     --attribute=gene_biotype:TR_C_gene
+
+
+    cellranger mkref --genome=GRCh38 \
+                     --fasta="$ref_fasta_filename" \
+                     --genes="filtered.$gtf_filename" \
+                     --nthreads=$(getconf _NPROCESSORS_ONLN)
+
+    tar cvf ${ref_name}.tar GRCh38/
```

Comment:
We could save some space and file transfer time by gzipping it. Is there a reason why we wouldn't want to?

Reply:
No, there are no special reasons for doing this, except:
- I thought using tar.gz will cause the cellranger run slower
- I copied some code from existing build reference wdls, which are using tars
I think we can use `tar cvzf` here!

Reply:
Hmm, yeah it will make cellranger run a little slower if it has to gunzip. (Copy time will be slightly faster but it seems it's outweighed by decompression time based on an old discussion in slack.) Maybe it's better to not gzip for now and we can add it later if we need to.

Reply:
Yea, we ran into the same problem while we were working with SS2.  `gunzip`  definitely takes some time

Reply:
If you do end up gzipping, consider `tar cvf - GRCh38/ | pigz > "${ref_name}.tar.gz"`

Reply:
The index files are binary files, I am wondering how much you can reduce in size if you compress.